Composer/packages/client/src/recoilModel/dispatchers/orchestrator.ts

```diff
@@ -33,46 +35,55 @@ export const downloadModel = (addr: string, model: 'en' | 'multilang') => {
   });
 };
 
-export const orchestratorDispatcher = () => {
-  const downloadLanguageModels = useRecoilCallback(
-    ({ set, snapshot }: CallbackInterface) => async (projectId: string) => {
-      const recognizers = await snapshot.getPromise(recognizersSelectorFamily(projectId));
+export const availableLanguageModels = (recognizerFiles: RecognizerFile[]) => {
+  const dialogsUsingOrchestrator = recognizerFiles?.filter(
+    ({ id, content }) => content.$kind === SDKKinds.OrchestratorRecognizer
+  );
+  const languageModels: ('en' | 'multilang')[] = [];
+  if (dialogsUsingOrchestrator.length) {
+    // pull out languages that Orchestrator has to support
+    const [enLuFiles, multiLangLuFiles] = partition(dialogsUsingOrchestrator, (f) =>
+      f.id.split('.')?.[1]?.toLowerCase()?.startsWith('en')
+    );
 
-      const isUsingOrchestrator = recognizers.some(
-        ({ id, content }) => content.$kind === SDKKinds.OrchestratorRecognizer
-      );
+    if (enLuFiles.length) {
+      languageModels.push('en');
+    }
 
-      if (isUsingOrchestrator) {
-        // pull out languages that Orchestrator has to support
-        const botLanguages = recognizers
-          .filter(({ id, content }) => content.$kind === SDKKinds.OrchestratorRecognizer)
-          .map(({ id, content }) => id.split('.')?.[1]);
+    if (
+      multiLangLuFiles
+        .map((r) => r.id.split('.')?.[1])
+        ?.some((lang) => Locales.map((l) => l.locale).includes(lang?.toLowerCase()))
+    ) {
+      languageModels.push('multilang');
+    }
+  }
+  return languageModels;
+};
 
-        const languageModels: ('en' | 'multilang')[] = [];
-        if (botLanguages.some((l) => l?.startsWith('en'))) {
-          languageModels.push('en');
-        }
+export const orchestratorDispatcher = () => {
+  const downloadLanguageModels = useRecoilCallback(({ snapshot }: CallbackInterface) => async (projectId: string) => {
+    const recognizers = await snapshot.getPromise(recognizersSelectorFamily(projectId));
 
-        // TODO: We need to get a table of Orchestrator supported languages
-        if (botLanguages.length > 1) {
-          languageModels.push('multilang');
-        }
+    const isUsingOrchestrator = recognizers?.some(
```

Comment:
nit: can recognizers not have value?

Reply:
Thanks @hatpick , addressed all your comments.  Yes, you're right the recognizer array will always have some value. Removed `?` from this line as well.